### PR TITLE
Testkit: introduce log capturing

### PR DESCRIPTION
**Pull request by ennru**
## Purpose

Introduce log capturing the with an Alpakka testkit which lends code from akka-actor-typed-testkit.

## References

https://doc.akka.io/docs/akka/current/typed/testing-async.html#silence-logging-output-from-tests

### CODE REVIEW

**Comment by ennru** [comment]
I have issues with proper initalisation of the Logback loggers before the Log Capture logging kicks in.

File: ftp/src/test/java/akka/stream/alpakka/ftp/CommonFtpStageTest.java [hunk 1]
```diff
@@ -38,6 +38,10 @@
   Sink<FtpFile, CompletionStage<IOResult>> getMoveSink(Function<FtpFile, String> destinationPath)
       throws Exception;
 
+  default IOResult await(CompletionStage<IOResult> result) throws InterruptedException, java.util.concurrent.ExecutionException, java.util.concurrent.TimeoutException {
+    return result.toCompletableFuture().get(5, TimeUnit.SECONDS);
```
Comment (ennru) [inline ftp/src/test/java/akka/stream/alpakka/ftp/CommonFtpStageTest.java]: Increased timeout for these Java tests.

File: google-cloud-pub-sub-grpc/src/test/resources/logback-test.xml [hunk 2]
```diff
@@ -1,11 +1,28 @@
 <configuration>
-    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+    <appender name="FILE" class="ch.qos.logback.core.FileAppender">
+        <file>target/google-cloud-pub-sub-grpc.log</file>
+        <append>false</append>
         <encoder>
             <pattern>%d{ISO8601} %-5level [%thread] [%logger{36}]  %msg%n</pattern>
         </encoder>
     </appender>
```
Comment (ennru) [inline google-cloud-pub-sub-grpc/src/test/resources/logback-test.xml]: This didn't create a log file at all.

File: s3/src/test/scala/akka/stream/alpakka/s3/impl/HttpRequestsSpec.scala [hunk 3]
```diff
@@ -397,28 +398,29 @@ class HttpRequestsSpec extends AnyFlatSpec with Matchers with ScalaFutures {
...
 
-    probe.expectMsgType[HttpRequest]
+      Http().singleRequest(req).futureValue shouldBe a[HttpResponse]
 
-    materializer.shutdown()
-    TestKit.shutdownActorSystem(system)
+      probe.expectMsgType[HttpRequest]
+    } finally {
+      TestKit.shutdownActorSystem(system)
+    }
```
Comment (ennru) [inline s3/src/test/scala/akka/stream/alpakka/s3/impl/HttpRequestsSpec.scala]: This fails sometimes when finding a port https://github.com/akka/alpakka/issues/1802

File: unix-domain-socket/src/main/scala/akka/stream/alpakka/unixdomainsocket/impl/UnixDomainSocketImpl.scala [hunk 4]
```diff
@@ -437,6 +437,13 @@ private[unixdomainsocket] abstract class UnixDomainSocketImpl(system: ExtendedAc
           }
         )
       } catch {
+        case e: IOException =>
+          val withAddress = new IOException(e.getMessage + s" ($address)", e)
+          registeredKey.cancel()
+          channel.close()
+          incomingConnectionQueue.fail(withAddress)
+          serverBinding.failure(withAddress)
```
Comment (ennru) [inline unix-domain-socket/src/main/scala/akka/stream/alpakka/unixdomainsocket/impl/UnixDomainSocketImpl.scala]: More information to understand why some tests keep failing.

File: ftp/src/test/java/akka/stream/alpakka/ftp/CommonFtpStageTest.java (hunk 1 above)
Comment (seglo) [reply ftp/src/test/java/akka/stream/alpakka/ftp/CommonFtpStageTest.java]: A build broke even with 5 seconds.  Maybe upgrade to 10 before merging?

https://travis-ci.com/akka/alpakka/jobs/277549627#L943

File: unix-domain-socket/src/test/java/docs/javadsl/UnixDomainSocketTest.java [hunk 5]
```diff
@@ -64,22 +72,31 @@ public void aUnixDomainSocketShouldReceiveWhatIsSent() throws Exception {
...
-                  .map(ByteString::fromString);
-
-          connection.handleWith(echo, materializer);
-        },
-        materializer);
+    CompletionStage<ServerBinding> streamCompletion =
+        connections
+            .map(
+                connection -> {
+                  System.out.println("New connection from: " + connection.remoteAddress());
```
Comment (seglo) [inline unix-domain-socket/src/test/java/docs/javadsl/UnixDomainSocketTest.java]: Could use a logger?

**Pull request merged**